### ISSUE TITLE
Add PEM format support in RSA algorithms.

### ISSUE THREAD

**Issue by lolgear** [yourkarma/JWT#121]
### New Issue Checklist

* [x] I have read and understood the [CONTRIBUTING guide](https://github.com/yourkarma/JWT/blob/master/.github/CONTRIBUTING.md)
* [x] I have read the [Documentation](https://github.com/yourkarma/JWT#documentation)
* [x] I have searched for a similar issue in the [project](https://github.com/yourkarma/JWT/issues) and found none

### Issue Info

 Info                    | Value                               |
-------------------------|-------------------------------------|
 Platform Name           | e.g. ios / osx / tvos / watchos
 Platform Version        | e.g. 8.0
 CocoaLumberjack Version | e.g. 2.3.0
 Integration Method      | e.g. carthage / cocoapods / manually
 Xcode Version           | e.g. Xcode 7.3
 Repro rate              | e.g. all the time (100%) / sometimes x% / only once
 Demo project link       | e.g. link to a demo project that highlights the issue

### Issue Description and Steps

Please fill in the detailed description of the issue (full output of any stack trace, compiler error, ...) and the steps to reproduce the issue.

It would be great to add PEM format support in RSA algorithms.

As a consequence, it could allow private/public key role swap ( public key signing or private key signing ).

* [ ] Add PEM format support.
* [ ] Add private/public key role swap.

**Comment by benjamindohertysc** [comment]
I'm confused about this issue- is RS256 with PEM formatted keys not already supported? The "Keys loaded from Pem files" example in the README seems to hint otherwise, unless I'm missing something.

**Comment by lolgear** [comment]
@benjamindohertysc it is only a blueprint.

In case of big plans ( EC support ), API could change.
Maybe it is better to hold EC support for version 3.1 or later.

**Comment by forbze** [comment]
I've spent the last few hours trying to read a PEM file in thinking it was implemented - I'd recommend putting a note on the README to explain its current state and tips to get it working.

Current state also does not support the ----BEGIN PRIVATE KEY---- / ----END PRIVATE KEY---- markers in the files. Once I removed those and hardcoded the public key in a string without the line breaks, everything worked as expected.

**Comment by lolgear** [comment]
@forbze try #148

**Comment by lolgear** [comment]
@forbze 
could it be closed?

**Comment by forbze** [comment]
Sorry - I haven’t had chance to test, but the commit looks like it would fix the problems. Happy to close my part.
